include/psa/crypto_values.h

```diff
@@ -1555,9 +1563,43 @@
  * This lifetime value is the default storage area for the calling
  * application. Implementations may offer other storage areas designated
  * by other lifetime values as implementation-specific extensions.
+ * See ::psa_key_lifetime_location_t for more information.
  */
 #define PSA_KEY_LIFETIME_PERSISTENT             ((psa_key_lifetime_t)0x00000001)
 
+#define PSA_KEY_LIFETIME_PERSISTENCE_VOLATILE ((psa_key_lifetime_persistence_t)0x00)
+#define PSA_KEY_LIFETIME_PERSISTENCE_PRIMARY ((psa_key_lifetime_persistence_t)0x01)
+#define PSA_KEY_LIFETIME_PERSISTENCE_READ_ONLY ((psa_key_lifetime_persistence_t)0xff)
+
+#define PSA_KEY_LIFETIME_GET_PERSISTENCE(lifetime)      \
+    ((psa_key_lifetime_persistence_t)((lifetime) & 0x000000ff)
+
+#define PSA_KEY_LIFETIME_GET_LOCATION(lifetime)      \
+    ((psa_key_lifetime_location_t)((lifetime) >> 8)
+
+/** Whether a key lifetime indicates that the key is volatile.
+ *
+ * A volatile key is automatically destroyed by the implementation when
+ * the application instance terminates. In particular, a volatile key
+ * is automatically on a power reset of the device.
+ *
+ * A key that is not volatile is persistent. Persistent keys are
```

Comment:
This means that there can't be a lifetime for keys that survives the termination of the application but not a power reset. I can hardly imagine a use case where that would be needed, I just would like to confirm that this is what we want?

Reply:
Seem sensible changes.
I am still thinking about "volatile" keys that last beyond the life of the calling partition.

Reply:
A lifetime that survives the instance of the calling application but not a power reset could exist, but:

* This is an imprecise definition. If the application is running in a virtual machine, is the scope of the power reset the VM or its physical host?
* How does another instance of the application refer to the same key? It would need to use a key identifier, so this would be more like a persistent key from the application's perspective.
* Yes, it's possible. I've worked with a system that basically had this third intermediate scope (volatile-to-application-instance, volatile-to-device, persistent). But AFAIK this intermediate scope was never used outside of the test suite. So I'm with “can hardly imagine a use case”.
* If an implementation really wants this feature, it can provide it through the vendor range for persistence levels.

Reply:
Yes that covers what I meant by "thinking".
In general PSA  has been writtien for M-class with a fixed set of secure partitions all of which active - so there really is no concept of application life time except reboot. 

However, when we extend to Cortex A we will need to deal with other life times - but I believe that can be left to the next release.